src/math/trig.c

```diff
@@ -94,9 +108,159 @@ void cordic_sin_cos(int32_t th_rad_fxp, cordic_cfg type, int32_t *sign, int32_t
 		xtmp = *xn >> (b_idx + 1);
 		ytmp = *b_yn >> (b_idx + 1);
 	}
-	/* Q2.30 format */
+	/* Q2.30 format -sine,cosine*/
 	*th_cdc_fxp = th_rad_fxp;
+}
+
+/**
+ * \CORDIC-based approximation for inverse sine, cosine
+ * Arguments	: int32_t realvalue
+ *		  int16_t numiters
+ *		  const int32_t LUT[CORDIC_30B_ITABLE_SIZE][30]
+ * Return Type	: int32_t
+ */
+int32_t is_scalar_cordic(int32_t realvalue, int16_t numiters,
+			 const int32_t LUT[CORDIC_30B_ITABLE_SIZE], cordic_cfg type)
+{
+	int32_t b_sincosvalue;
+	int32_t b_i;
+	int32_t i;
+	int32_t x = 0;
+	int32_t xdshift;
+	int32_t xshift;
+	int32_t y = 0;
+	int32_t ydshift;
+	int32_t yshift;
+	int32_t z = 0;
+	int16_t j;
+	int16_t k;
+
+	b_sincosvalue = (realvalue >> 1U);
+	if (type == EN_32B_CORDIC_ASINE) {
+		/* Initialize the variables for the cordic iteration
+		 * angles less than pi/4, we initialize (x,y) along the x-axis.
+		 * angles greater than or equal to pi/4, we initialize (x,y)
+		 * along the y-axis. This improves the accuracy of the algorithm
+		 * near the edge of the domain of convergence
+		 */
+		if (b_sincosvalue > cord_arcsincos_q28fl) {
+			x = 0;
+			y = cord_arcsincos_q30fl;
+			z = 843314856;
+		} else {
+			x = cord_arcsincos_q30fl;
+			y = 0;
+			z = 0;
+		}
+	}
+	if (type == EN_32B_CORDIC_ACOSINE) {
+		/* Initialize the variables for the cordic iteration
+		 * angles less than pi/4, we initialize (x,y) along the x-axis.
+		 * angles greater than or equal to pi/4, we initialize (x,y)
+		 * along the y-axis. This improves the accuracy of the algorithm
+		 * near the edge of the domain of convergence
+		 */
+		if (b_sincosvalue < cord_arcsincos_q28fl) {
+			x = 0;
+			y = cord_arcsincos_q30fl;
+			z = 843314856;
+		} else {
+			x = cord_arcsincos_q30fl;
+			y = 0;
+			z = 0;
+		}
+	}
+	b_sincosvalue <<= 1U;
+	/* DCORDIC(Double CORDIC) algorithm */
+	/* Double iterations method consists in the fact that unlike the classical */
+	/* CORDIC method,where the iteration step value changes EVERY time, i.e. on */
+	/* each iteration, in the double iteration method, the iteration step value */
+	/* is repeated twice and changes only through one iteration */
+	i = numiters - 1;
+	for (b_i = 0; b_i < i; b_i++) {
+		j = (int16_t)((b_i + 1) << 1U);
```

Comment:
If you don't need 16 bit overflows here the code will be faster if you keep everything 32 bit int. Xtensa is faster with native 32 bit arithmetic. (exception 16 bit 4x SIMD but it's not used here)